reconciler/reconcile_common.go

```diff
@@ -32,6 +32,8 @@ func PreProcessReconcile(ctx context.Context, resource duckv1.KRShaped) {
 	if newStatus.ObservedGeneration != resource.GetGeneration() {
 		condSet := resource.GetConditionSet()
 		manager := condSet.Manage(newStatus)
+		// Ensure conditions are initialized before we modify.
+		manager.InitializeConditions()
```

Comment:
Just to confirm my understanding, this behavior is opt-in per controller?

Reply:
Synced with @whaught online. This is opt-in and only a few controllers are using it. Those controllers already initialize conditions at the start of reconcile.

Reply:
https://github.com/knative/serving/pull/8169

Tested in serving to make sure.